### ISSUE TITLE
`(require 'helm-locate)` causes Lisp error: (void-variable it)

### ISSUE THREAD

**Issue by NateEag** [emacs-helm/helm#823]
I just updated my packages, and I now get the above error when starting emacs.

Here's a minimal recipe for reproducing, assuming you keep ELPA packages under `~/.emacs.d`:

```bash
$ cd ~/.emacs.d
$ emacs -Q
```

then in the Emacs instance

```elisp
M-S-:
Eval: (normal-top-level-add-subdirs-to-load-path) RET
M-S-:
Eval: (require 'helm-locate) RET
```

Result: the following stack trace is displayed:

```elisp
Debugger entered--Lisp error: (void-variable it)
  helm-source([object helm-locate-source "Locate" "Locate" nil helm-locate-set-command nil nil nil nil helm-generic-files-map (("Find file" . helm-find-many-files) ("Find file as root" . helm-find-file-as-root) ("Find file other window" . find-file-other-window) ("Find file other frame" . find-file-other-frame) ("Open dired in file's directory" . helm-open-dired) ("Grep File(s) `C-u recurse'" . helm-find-files-grep) ("Zgrep File(s) `C-u Recurse'" . helm-ff-zgrep) ("Pdfgrep File(s)" . helm-ff-pdfgrep) ("Insert as org link" . helm-files-insert-as-org-link) ("Checksum File" . helm-ff-checksum) ("Ediff File" . helm-find-files-ediff-files) ("Ediff Merge File" . helm-find-files-ediff-merge-files) ("Etags `M-., C-u tap, C-u C-u reload tag file'" . helm-ff-etags-select) ("View file" . view-file) ("Insert file" . insert-file) ("Add marked files to file-cache" . helm-ff-cache-add-file) ("Delete file(s)" . helm-delete-marked-files) ("Open file externally (C-u to choose)" . helm-open-file-externally) ("Open file with default tool" . helm-open-file-with-default-tool) ("Find file in hex dump" . hexl-find-file)) nil "Show this file" helm-generic-file-help-message nil 3 (helm-skip-boring-files helm-highlight-files helm-w32-pathname-transformer) nil nil nil nil (helm-transform-file-load-el helm-transform-file-browse-url helm-transform-file-cache) helm-locate-pattern-transformer 9999 nil nil nil nil nil nil nil helm-file-name-history nil helm-generic-file-mode-line-string nil nil nil nil (helm-compile-source--persistent-help) nil nil nil nil nil helm-locate-init])
  apply(helm-source [object helm-locate-source "Locate" "Locate" nil helm-locate-set-command nil nil nil nil helm-generic-files-map (("Find file" . helm-find-many-files) ("Find file as root" . helm-find-file-as-root) ("Find file other window" . find-file-other-window) ("Find file other frame" . find-file-other-frame) ("Open dired in file's directory" . helm-open-dired) ("Grep File(s) `C-u recurse'" . helm-find-files-grep) ("Zgrep File(s) `C-u Recurse'" . helm-ff-zgrep) ("Pdfgrep File(s)" . helm-ff-pdfgrep) ("Insert as org link" . helm-files-insert-as-org-link) ("Checksum File" . helm-ff-checksum) ("Ediff File" . helm-find-files-ediff-files) ("Ediff Merge File" . helm-find-files-ediff-merge-files) ("Etags `M-., C-u tap, C-u C-u reload tag file'" . helm-ff-etags-select) ("View file" . view-file) ("Insert file" . insert-file) ("Add marked files to file-cache" . helm-ff-cache-add-file) ("Delete file(s)" . helm-delete-marked-files) ("Open file externally (C-u to choose)" . helm-open-file-externally) ("Open file with default tool" . helm-open-file-with-default-tool) ("Find file in hex dump" . hexl-find-file)) nil "Show this file" helm-generic-file-help-message nil 3 (helm-skip-boring-files helm-highlight-files helm-w32-pathname-transformer) nil nil nil nil (helm-transform-file-load-el helm-transform-file-browse-url helm-transform-file-cache) helm-locate-pattern-transformer 9999 nil nil nil nil nil nil nil helm-file-name-history nil helm-generic-file-mode-line-string nil nil nil nil (helm-compile-source--persistent-help) nil nil nil nil nil helm-locate-init])
  eieio-generic-call(helm--setup-source ([object helm-locate-source "Locate" "Locate" nil helm-locate-set-command nil nil nil nil helm-generic-files-map (("Find file" . helm-find-many-files) ("Find file as root" . helm-find-file-as-root) ("Find file other window" . find-file-other-window) ("Find file other frame" . find-file-other-frame) ("Open dired in file's directory" . helm-open-dired) ("Grep File(s) `C-u recurse'" . helm-find-files-grep) ("Zgrep File(s) `C-u Recurse'" . helm-ff-zgrep) ("Pdfgrep File(s)" . helm-ff-pdfgrep) ("Insert as org link" . helm-files-insert-as-org-link) ("Checksum File" . helm-ff-checksum) ("Ediff File" . helm-find-files-ediff-files) ("Ediff Merge File" . helm-find-files-ediff-merge-files) ("Etags `M-., C-u tap, C-u C-u reload tag file'" . helm-ff-etags-select) ("View file" . view-file) ("Insert file" . insert-file) ("Add marked files to file-cache" . helm-ff-cache-add-file) ("Delete file(s)" . helm-delete-marked-files) ("Open file externally (C-u to choose)" . helm-open-file-externally) ("Open file with default tool" . helm-open-file-with-default-tool) ("Find file in hex dump" . hexl-find-file)) nil "Show this file" helm-generic-file-help-message nil 3 (helm-skip-boring-files helm-highlight-files helm-w32-pathname-transformer) nil nil nil nil (helm-transform-file-load-el helm-transform-file-browse-url helm-transform-file-cache) helm-locate-pattern-transformer 9999 nil nil nil nil nil nil nil helm-file-name-history nil helm-generic-file-mode-line-string nil nil nil nil (helm-compile-source--persistent-help) nil nil nil nil nil helm-locate-init]))
  helm--setup-source([object helm-locate-source "Locate" "Locate" nil helm-locate-set-command nil nil nil nil helm-generic-files-map (("Find file" . helm-find-many-files) ("Find file as root" . helm-find-file-as-root) ("Find file other window" . find-file-other-window) ("Find file other frame" . find-file-other-frame) ("Open dired in file's directory" . helm-open-dired) ("Grep File(s) `C-u recurse'" . helm-find-files-grep) ("Zgrep File(s) `C-u Recurse'" . helm-ff-zgrep) ("Pdfgrep File(s)" . helm-ff-pdfgrep) ("Insert as org link" . helm-files-insert-as-org-link) ("Checksum File" . helm-ff-checksum) ("Ediff File" . helm-find-files-ediff-files) ("Ediff Merge File" . helm-find-files-ediff-merge-files) ("Etags `M-., C-u tap, C-u C-u reload tag file'" . helm-ff-etags-select) ("View file" . view-file) ("Insert file" . insert-file) ("Add marked files to file-cache" . helm-ff-cache-add-file) ("Delete file(s)" . helm-delete-marked-files) ("Open file externally (C-u to choose)" . helm-open-file-externally) ("Open file with default tool" . helm-open-file-with-default-tool) ("Find file in hex dump" . hexl-find-file)) nil "Show this file" helm-generic-file-help-message nil 3 (helm-skip-boring-files helm-highlight-files helm-w32-pathname-transformer) nil nil nil nil (helm-transform-file-load-el helm-transform-file-browse-url helm-transform-file-cache) helm-locate-pattern-transformer 9999 nil nil nil nil nil nil nil helm-file-name-history nil helm-generic-file-mode-line-string nil nil nil nil (helm-compile-source--persistent-help) nil nil nil nil nil helm-locate-init])
  helm-make-source("Locate" helm-locate-source :pattern-transformer helm-locate-pattern-transformer)
  (defvar helm-source-locate (helm-make-source "Locate" (quote helm-locate-source) :pattern-transformer (quote helm-locate-pattern-transformer)))
  require(helm-locate)
  eval((require (quote helm-locate)) nil)
  eval-expression((require (quote helm-locate)) nil)
  call-interactively(eval-expression nil nil)
  command-execute(eval-expression)
```

I'm on  `helm-20150114.140` from MELPA, installed via package.el. Desktop is OS X 10.10.1, and `M-x emacs-version` says `GNU Emacs 24.4.1 (x86_64-apple-darwin13.4.0, NS apple-appkit-1265.21) of 2014-10-20 on builder10-9.porkrind.org`.

**Comment by michael-heerdegen** [comment]
FWIW I can't reproduce this here.

**Comment by thierryvolpiatto** [comment]
Nate Eagleson <notifications@github.com> writes:

> I just updated my packages, and I now get the above error when starting emacs.
>
> Here's a minimal recipe for reproducing, assuming you keep ELPA packages under ~/.emacs.d:
>
> $ cd ~/.emacs.d
> $ emacs -Q
>
> then in the Emacs instance

PACKAGE-INITIALIZE

> M-S-:
> Eval: (normal-top-level-add-subdirs-to-load-path) RET

NO!

> M-S-:
> Eval: (require 'helm-locate) RET

Maybe but preferably use (require-helm-config)

Please read the documentation of package.el and the readme file of helm.

-- 
Thierry
Get my Gnupg key:
gpg --keyserver pgp.mit.edu --recv-keys 59F29997

**Comment by NateEag** [comment]
I use package-initialize to set up package.el in my config.

The above was an attempt to reproduce the issue without loading any unrelated config, since that's what other maintainers I've worked with usually want.

My apologies for not reading the Helm readme before filing this report.

@michael-heerdegen: I can't reproduce this on my home machine even with my invalid recipe, so there's some difference between these machines.

I'll try reproducing according to the docs on my work machine, to make sure this issue isn't valid.

**Comment by NateEag** [comment]
I had reverted my work machine to an older helm build after filing this report so I could keep working. 

After moving back to the current version, this issue no longer occurs. I'm not sure what caused it, but it clearly was something wrong with my setup. I suspect it might have to do with my post-merge githook to byte-compile files (something I set up before I knew about [auto-compile](https://github.com/tarsius/auto-compile).

My apologies for the bad report.
